file-manager/src/cli.js

```diff
@@ -0,0 +1,159 @@
+  const commandResult = OsOps.handleCommand(args[0]);
+  if (commandResult) {
+    process.stdout.write(`${commandResult}${EOL}`);
+  } else {
+    process.stdout.write(invalidInputMessage);
+  }
+};
+const handleHash = async (args) => {
+  const hash = await HashOps.getHash(args[0]);
+  process.stdout.write(`${hash}${EOL}`);
+}
+const handlerCompress = async (args) => {
+  await CompressOps.compress(args[0], args[1]);
+}
+const handlerDecompress = async (args) => {
+  await CompressOps.decompress(args[0], args[1]);
+}
+
+const commandHandlers = {
+  ['.exit']: handleExit,
+  ['up']: handleUp,
+  ['cd']: handleChangeDirectory,
+  ['ls']: handleList,
+  ['cat']: handleConcatenation,
+  ['add']: handleAdd,
+  ['rn']: handleRename,
+  ['cp']: handleCopy,
+  ['mv']: handleMove,
+  ['rm']: handleRemove,
+  ['os']: handleOsCommand,
+  ['hash']: handleHash,
+  ['compress']: handlerCompress,
+  ['decompress']: handlerDecompress,
+};
+
+const printCurrentDirectory = () => {
+  process.stdout.write(`You are currently in ${process.cwd()}${EOL}`);
+}
+
+const handleUserInput = async (input) => {
+  // TODO: In case of unknown operation or invalid input
+  // (missing mandatory arguments, wrong data in arguments, etc.) Invalid input message should be shown
+  const [command, ...args] = input
+    .match(whiteSpaceSplitRegex)
+    .map(arg => arg.replaceAll(quoteRegex, ''));
+  const handler = commandHandlers[command];
+
+  if (handler) {
+    await handler(args);
+  } else {
+    process.stdout.write(invalidInputMessage);
+  }
+
+  printCurrentDirectory();
+};
+
+const setName = (args) => {
+  const argumentsEntries = args.map(entry => entry.split('='));
+  const userNameEntry = argumentsEntries.find(entry => entry[0] === '--username');
+  if (!userNameEntry) {
+    throw new Error(`name is not provided`);
+  }
+
+  userName = userNameEntry[1] || 'Incognito';
+};
+
+const setProcessHandlers = () => {
+  process.on('SIGINT', () => {
+    process.exit(0);
+  });
+
+  process.on('SIGTERM', () => {
+    process.exit(0);
+  })
+
+  process.on('error', (error) => {
+    console.error(error)
+    process.stdout.write(operationFailedMessage);
+  });
+
+  process.on('unhandledRejection', (error) => {
+    console.error(error)
+    process.stdout.write(operationFailedMessage);
+  });
+
+  process.on('uncaughtException', (error) => {
+    if (error?.message === 'name is not provided') {
+      process.stdout.write(nameNotProvidedMessage);
+      process.exit(1);
+    } else {
+      console.error(error)
+      process.stdout.write(operationFailedMessage);
+    }
+  });
+
+  process.on('exit', (code) => {
+    if (code === 0) {
+      process.stdout.write(`Thank you for using File Manager, ${userName}, goodbye!${EOL}`);
+    }
+  });
+};
+
+export const start = (args) => {
+  setProcessHandlers();
+  setName(args);
+  process.chdir(homedir());
+  printCurrentDirectory();
+  process.stdout.write(`Welcome to the File Manager, ${userName}!${EOL}`);
```

Comment:
NOTE: it might be more convenient to implement it via readline interface to have ability to have commands history and navigate through them via 'up' and 'down' arrows